base/path.jl

```diff
@@ -477,3 +477,73 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isabspath, :isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+
+module FOLDERID
+
+struct GUID
+    l1::Culong
+    w1::Cushort
+    w2::Cushort
+    b::NTuple{8,Cuchar}
+end
+
+const Fonts                  = GUID(0xFD228CB7, 0xAE11, 0x4AE3, (0x86, 0x4C, 0x16, 0xF3, 0x91, 0x0A, 0xB8, 0xFE))
+const Desktop                = GUID(0xB4BFCC3A, 0xDB2C, 0x424C, (0xB0, 0x29, 0x7F, 0xE9, 0x9A, 0x87, 0xC6, 0x41))
+const System                 = GUID(0x1AC14E77, 0x02E7, 0x4E5D, (0xB7, 0x44, 0x2E, 0xB1, 0xAE, 0x51, 0x98, 0xB7))
+const SystemX86              = GUID(0xD65231B0, 0xB2F1, 0x4857, (0xA4, 0xCE, 0xA8, 0xE7, 0xC6, 0xEA, 0x7D, 0x27))
+const Windows                = GUID(0xF38BF404, 0x1D43, 0x42F2, (0x93, 0x05, 0x67, 0xDE, 0x0B, 0x28, 0xFC, 0x23))
+const Profile                = GUID(0x5E6C858F, 0x0E22, 0x4760, (0x9A, 0xFE, 0xEA, 0x33, 0x17, 0xB6, 0x71, 0x73))
+const ProgramFilesX86        = GUID(0x7C5A40EF, 0xA0FB, 0x4BFC, (0x87, 0x4A, 0xC0, 0xF2, 0xE0, 0xB9, 0xFA, 0x8E))
+const ProgramFilesCommonX86  = GUID(0xDE974D24, 0xD9C6, 0x4D3E, (0xBF, 0x91, 0xF4, 0x45, 0x51, 0x20, 0xB9, 0x17))
+const ProgramFilesX64        = GUID(0x6d809377, 0x6af0, 0x444b, (0x89, 0x57, 0xa3, 0x77, 0x3f, 0x02, 0x20, 0x0e))
+const ProgramFilesCommonX64  = GUID(0x6365d5a7, 0xf0d , 0x45e5, (0x87, 0xf6, 0xd , 0xa5, 0x6b, 0x6a, 0x4f, 0x7d))
+const ProgramFiles           = GUID(0x905e63b6, 0xc1bf, 0x494e, (0xb2, 0x9c, 0x65, 0xb7, 0x32, 0xd3, 0xd2, 0x1a))
+const ProgramFilesCommon     = GUID(0xF7F1ED05, 0x9F6D, 0x47A2, (0xAA, 0xAE, 0x29, 0xD3, 0x17, 0xC6, 0xF0, 0x66))
+const UserProgramFiles       = GUID(0x5cd7aee2, 0x2219, 0x4a67, (0xb8, 0x5d, 0x6c, 0x9c, 0xe1, 0x56, 0x60, 0xcb))
+const UserProgramFilesCommon = GUID(0xbcbd3057, 0xca5c, 0x4622, (0xb4, 0x2d, 0xbc, 0x56, 0xdb, 0x0a, 0xe5, 0x16))
+const RoamingAppData         = GUID(0x3EB685DB, 0x65F9, 0x4CF6, (0xA0, 0x3A, 0xE3, 0xEF, 0x65, 0x72, 0x9F, 0x3D))
+const LocalAppData           = GUID(0xF1B32785, 0x6FBA, 0x4FCF, (0x9D, 0x55, 0x7B, 0x8E, 0x7F, 0x15, 0x70, 0x91))
+const LocalAppDataLow        = GUID(0xA520A1A4, 0x1780, 0x4FF6, (0xBD, 0x18, 0x16, 0x73, 0x43, 0xC5, 0xAF, 0x16))
+
+end
+
+if Sys.iswindows()
+
+import .FOLDERID
+import .FOLDERID: GUID
+
+const KF_FLAG_DEFAULT = 0x00000000
+
+
+"""
+    get_known_folder_path(folderid::GUID) -> AbstractString
+
+Windows only function to retrieve the full path of a known folder identified by the folder's FOLDERID.
+The FOLDERID module contains the folderid constants available for query.
+
+# Examples
+
+```
+import Base.Filesystem: get_known_folder_path
+import Base.Filesystem.FOLDERID
+
+julia> get_known_folder_path(FOLDERID.System)
+"C:\\WINDOWS\\system32"
+
+julia> get_known_folder_path(FOLDERID.Fonts)
+"C:\\WINDOWS\\Fonts"
+```
+
+"""
+function get_known_folder_path(folderid::GUID)
+    pathptr = Ref{Ptr{Cwchar_t}}()
+    result = ccall((:SHGetKnownFolderPath,:shell32), stdcall, Cint,
```

Comment:
For some reason I am unable to get this line to run on win-32bit.  It keeps spitting `ERROR: ReadOnlyMemoryError()`